### ISSUE TITLE
Home Assistant as a geolocation data source

### ISSUE THREAD

**Issue by rubiojr** [timelinize/timelinize#21]
Devices with the Home Assistant companion app installed (like phones) can send geolocation data back to Home Assistant that can be used for tracking purposes (see https://github.com/rubiojr/hass2geo).

We can probably build a data source that listens to websocket or MQTT events in Home Assistant and ingests those geolocation events from devices.

**Comment by mholt** [comment]
Good idea. Though I haven't really designed "push" ingestion yet. I suppose we need some sort of API to add items to the timeline.

Or, if there's a way to get that data out of HA in bulk, that could be a good shorter-term solution.

**Comment by rubiojr** [comment]
> Though I haven't really designed "push" ingestion yet. I suppose we need some sort of API to add items to the timeline.

That'd be sweet I think, would open the door for people to build their own ingestors without having to add them to the list of Timelinize core datasources.

**Comment by mholt** [comment]
@rubiojr I made https://github.com/timelinize/timelinize/issues/31 to track an API design.
